### PR TITLE
Fix INCLUDEUSER flag

### PR DESCRIPTION
**Pull request by clusterfack**
Before_cast wasn't finding the user for some reason and returning it as a valid target.

### CODE REVIEW

File: code/modules/spells/aoe_turf/fall.dm [hunk 1]
```diff
@@ -72,7 +72,7 @@
 		invocation(user, targets)
 		take_charge(user, skipcharge)
 
-		targets = before_cast(targets)
+		targets = before_cast(targets, user)
```
Comment (ComicIronic) [inline code/modules/spells/aoe_turf/fall.dm:75]: Why does this spell copy perform? Is it literally just for the invocation snowflake which should have been done in invocation()

**Comment by ComicIronic** [comment]
You should not have self merged this.

File: code/modules/spells/spell_code.dm [hunk 2]
```diff
@@ -247,6 +248,8 @@ var/list/spells = typesof(/spell) //needed for the badmin verb for now
 			spawn(overlay_lifespan)
 				qdel(spell)
 				spell = null
+	if(spell_flags & INCLUDEUSER)
```
Comment (ComicIronic) [inline code/modules/spells/spell_code.dm:251]: This is going to cause issues. Not only does this flag overlap with IGNOREDENSE for the reason that it's only meant to be used in /targeted spells, it also adds the user to lists which should only include turfs for /aoe_turf spells.

File: code/modules/spells/aoe_turf/fall.dm (hunk 1 above)
Comment (clusterfack) [inline code/modules/spells/aoe_turf/fall.dm]: I don't know why it copies perform, I recently nixed another perform() version myself in my last pr

File: code/modules/spells/spell_code.dm (hunk 2 above)
Comment (clusterfack) [inline code/modules/spells/spell_code.dm]: /aoe_turf spells override this, I am all too aware of how all these systems work now

Comment (ComicIronic) [reply code/modules/spells/spell_code.dm:251]: Then what case is this fixing?

Comment (clusterfack) [inline code/modules/spells/spell_code.dm]: Spell/targeted/ethereal_jaunt

**Pull request merged**